src/controllers/authenticationController.ts

```diff
@@ -105,6 +106,70 @@ export default class AuthenticationController {
       .send(metadata);
   }
 
+  /**
+   * Returns data about the current user session.
+   */
+  public getSessionState(req: express.Request, res: express.Response): void {
+    if (req.headers && req.headers.authorization) {
+      const authorization = req.headers.authorization as string;
+      const parts = authorization.split(" ");
+      if (parts.length === 2) {
+        const scheme = parts[0];
+        const token = parts[1];
+
+        if (/^Bearer$/i.test(scheme)) {
+          this.sessionStorage
```

Comment:
this function will be much more readable if you extract this logic in an async function (e.g. `refreshUserToken`) that returns a `Promise<Either<Error, string>>`